windows_cygwin/build.bat

```diff
@@ -1,5 +1,19 @@
-choco install -y firefox googlechrome wget
+@echo on
+choco install -y firefox googlechrome wget notepadplusplus sublimetext3
+
 wget -c https://cygwin.com/setup-x86_64.exe
-REM cygwin installation
-setup-x86_64.exe -R "C:\cygwin64" -s http://cygwin.netbet.org/ -q -g -P curl,zsh,git,vim,wget,xz,tar,gawk,bzip2,subversion
+setup-x86_64.exe -R "C:\cygwin64" -s http://cygwin.netbet.org/ -q -g -P curl,zsh,git,vim,wget,xz,tar,gawk,bzip2,subversion,zlib,fontconfig
+
+set bash=c:\cygwin64\bin\bash.exe --login -c
+for /f "delims=" %%A in ('%bash% "cd `cygpath $HOMEPATH`/dev/dotfiles && git rev-parse --abbrev-ref HEAD"') do set "branch=%%A" 
+%bash% 'echo "branch is $branch"'
```

Comment:
seems like these could be all in one script in the repo and invoked by oneliner here